Jenkinsfile

```
@@ -11,35 +26,12 @@ build('woody_erlang', 'docker-host', finalHook) {
   checkoutRepo()
   loadBuildUtils()
 
-  def pipeDefault
-  def withWsCache
+  def pipeErlangLib
   runStage('load pipeline') {
     env.JENKINS_LIB = "build_utils/jenkins_lib"
-    pipeDefault = load("${env.JENKINS_LIB}/pipeDefault.groovy")
-    withWsCache = load("${env.JENKINS_LIB}/withWsCache.groovy")
+    env.SH_TOOLS = "build_utils/sh"
+    pipeErlangLib = load("${env.JENKINS_LIB}/pipeErlangLib.groovy")
   }
 
-  pipeDefault() {
-    runStage('compile') {
-      withGithubPrivkey {
-        sh 'make wc_compile'
-      }
-    }
-    runStage('lint') {
-      sh 'make wc_lint'
-    }
-    runStage('xref') {
-      sh 'make wc_xref'
-    }
-    runStage('dialyze') {
-      withGithubPrivkey {
-        withWsCache("_build/test/rebar3_23.0.1_plt") {
-          sh 'make wc_dialyze'
-        }
-      }
-    }
-    runStage('test') {
-      sh "make wc_test"
-    }
-  }
+  pipeErlangLib.runPipe(false,false,'test')
```

Не пора ли уже здесь использовать [именованные параметры функции](http://docs.groovy-lang.org/latest/html/documentation/#_named_parameters_2)?

---

Предлагаю заводить их тут, если надо будет ещё добавить параметров и/или что-то поменять. Пока вроде трёх достаточно

---

Я имел ввиду что-то типа:

```suggestion
  pipeErlangLib.runPipe(testWithDependencies: false, runInParallel: false, dialyzeWithProfile:'test')
```

Как я понимаю, синтаксис groovy такое позволяет.

---

сделал

---

а нет, так нельзя, лол откатил